MaterialRemapper/Editor/MaterialRemapperView.cs

```diff
@@ -0,0 +1,246 @@
+                                {
+                                    // mesh list to map to materials
+                                    EditorGUILayout.BeginVertical();
+                                    {
+                                        GUILayout.Label(MaterialRemapperLocalization.remapMesh, EditorStyles.boldLabel, GUILayout.Width(500));
+
+                                        // list the objects
+                                        foreach (var remapItem in model.remapList)
+                                        {
+                                            EditorGUILayout.BeginHorizontal();
+
+                                            // mesh that we want to remap
+                                            GUILayout.Label(remapItem.mesh.name, GUILayout.Width(200));
+
+                                            // show all of the materials in the mesh
+                                            for (var j = 0; j < remapItem.mesh.sharedMaterials.Length; j++)
+                                            {
+                                                // do we have any materials to remap
+                                                remapItem.selectedIndex[j] = EditorGUILayout.Popup(remapItem.selectedIndex[j], model.materialList.Select(x=>x.name).ToArray(), GUILayout.Width(200));
+                                            }
+
+                                            EditorGUILayout.EndHorizontal();
+                                        }
+                                    }
+                                    EditorGUILayout.EndVertical();
+                                }
+                                EditorGUILayout.EndHorizontal();
+                            }
+                            else
+                            {
+                                GUILayout.Label(MaterialRemapperLocalization.noMaterialToRemap, EditorStyles.boldLabel, GUILayout.Width(500));
+                            }
+                        }
+                        if (showGlobalUI)
+                        {
+                            GUILayout.FlexibleSpace();
+                            if (GUILayout.Button("Apply Material Changes", GUILayout.Width(300), GUILayout.Height(35)))
+                            {
+                                model.ApplyMaterialChanges();
+                            }
+                        }
+                        GUILayout.Space(20);
+                    }
+                }
+                EditorGUILayout.EndVertical();
+            }
+            EditorGUILayout.EndScrollView();
+        }
+        
+        
+        
+        /// <summary>
+        /// Localization table for the MaterialRemapperView class
+        /// </summary>
+        static class MaterialRemapperLocalization
+        {
+            // UI Labels
+            public const string menuItemLabel = "Tools/Edit/Material Remapper";

```

Comment:
Window > General > Film/TV toolbox > Material Remapper ?

Reply:
I do not like heavily nested menus, especially when this would be the only entry. 
How about: Window > General > Material Remapper ?